### PR TITLE
Moving usage models to common module for reusability

### PR DESCRIPTION
**Pull request by mkuzdowicz**
## What does this change?
Moving usage models to common module for reusability

We need to move Usage models to common module to be able to re-project all the image additionally data that are stored in DynamoDB

like:
- leases 
- collections
- usages
- metadata (Edits)

for leases, collections and edits all is accessible from common module
What left behind is usage model

To be able to have an endpoint that will project how leases, usages, collections and edits for a given image should look like

we need to have their models in common module

## How can success be measured?

all work as it used to

This is code structure change not a feature change

## Tested?
- [x] locally
- [x] on TEST

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import com.gu.mediaservice.model.usage._
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object ItemToMediaUsage {
+
+  def transform(item: Item) = {
```
Comment (LATaylor-guardian) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala]: Can this have a response type please given it's a public function.

File: common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    }.toOption
+  }
+
+  private def buildPrint(metadataMap: Map[String, Any]): Option[PrintUsageMetadata] = {
+    type JStringNumMap = java.util.LinkedHashMap[String, java.math.BigDecimal]
+    Try {
+      PrintUsageMetadata(
+        sectionName = metadataMap.apply("sectionName").asInstanceOf[String],
+        issueDate = metadataMap.get("issueDate").map(_.asInstanceOf[String])
+          .map(ISODateTimeFormat.dateTimeParser().parseDateTime).get,
```
Comment (LATaylor-guardian) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala]: is the `.get` safe here?

Comment (LATaylor-guardian) [reply common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala]: Ah... suppose it's surrounded by a Try.

File: common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala (hunk 1 above)
Comment (mkuzdowicz) [reply common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala]: sure

File: common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala (hunk 2 above)
Comment (mkuzdowicz) [reply common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala]: I haven't put much attention to details as most of it is moving code around as building blocks not in details what the code does.

I did not introduced some new implementations, i mainly change the structure

File: common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala (hunk 1 above)
Comment (mkuzdowicz) [reply common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala]: done

File: common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala (hunk 2 above)
Comment (mkuzdowicz) [reply common-lib/src/main/scala/com/gu/mediaservice/lib/usage/ItemToMediaUsage.scala]: that function was present before as MediaUsageOps.build: https://github.com/guardian/grid/blob/master/usage/app/model/MediaUsage.scala#L44

File: common-lib/src/main/scala/com/gu/mediaservice/model/leases/MediaLease.scala [hunk 3]
```diff
@@ -10,6 +10,12 @@ sealed trait MediaLeaseType { def name: String }
 object MediaLeaseType {
   implicit val reads: Reads[MediaLeaseType] = {
     JsPath.read[String].map {
+      case "AllowUseLease"          => AllowUseLease
```
Comment (akash1810) [inline common-lib/src/main/scala/com/gu/mediaservice/model/leases/MediaLease.scala]: These changes feel out of scope for this PR as it changes how we serialise/deserialise. Can it be moved into it's own PR?

File: usage/app/model/MediaUsage.scala [hunk 4]
```diff
@@ -1,131 +0,0 @@
-package model
```
Comment (akash1810) [inline usage/app/model/MediaUsage.scala]: can we `git mv` this file to keep the history?

File: common-lib/src/main/scala/com/gu/mediaservice/model/leases/MediaLease.scala (hunk 3 above)
Comment (mkuzdowicz) [reply common-lib/src/main/scala/com/gu/mediaservice/model/leases/MediaLease.scala]: you are correct :)  thanks
But i will need to add them in next PR

Comment (mkuzdowicz) [reply common-lib/src/main/scala/com/gu/mediaservice/model/leases/MediaLease.scala]: done

File: usage/app/model/MediaUsage.scala (hunk 4 above)
Comment (mkuzdowicz) [reply usage/app/model/MediaUsage.scala]: done it, but not sure that PR is showing that correctly
I divide it to 3 commits to reflect that

**Comment by jonathonherbert** [comment]
This LGTM -- @akash1810 is there anything else to address here?
